src/js/libgeo/GeoOps.js

```diff
@@ -1928,14 +2018,23 @@ geoMacros.Calculation = function(el) {
     return [];
 };
 
+geoMacros.Arc = function(el) {
+    el.type = "ArcBy3";
+    return [el];
+};
+
 geoMacros.Transform = function(el) {
-    var trKind = csgeo.csnames[el.args[0]].kind;
-    var argKind = csgeo.csnames[el.args[1]].kind;
+    var arg = csgeo.csnames[el.args[1]];
+    var tr = csgeo.csnames[el.args[0]];
+    // workaround for Arcs since we treat them as circles
+    var akind = arg.kind;
+    if (arg.kind === "C" && arg.type === "ArcBy3") akind = "Arc";
```

Comment:
Moving to geoOps isn't a bad idea, but we would have to keep the
el.isArc property since GeoRender.js will only have the element handed
over and can't distinguish circles and arcs anymore in case we remove this.

Perhaps our best shot is #144. I already read the code, but i still have
to run some examples.
 
On 12/12/2015 04:04 PM, Martin von Gagern wrote:
>>  geoMacros.Transform = function(el) {
>> -    var trKind = csgeo.csnames[el.args[0]].kind;
>> -    var argKind = csgeo.csnames[el.args[1]].kind;
>> +    var arg = csgeo.csnames[el.args[1]];
>> +    var tr = csgeo.csnames[el.args[0]];
>> +    // workaround for Arcs since we treat them as circles
>> +    var akind = arg.kind;
>> +    if (arg.kind === "C" && arg.type === "ArcBy3") akind = "Arc";
>> … the macros are evaluated before initialization takes place.
> Then how about moving the `isArc` property from the geometric element to the geometric operator? So you can do `geoOps[arg.type].isArc`. There definitely should be a test case for repeated transformations, since this is more tricky than it would seem at first glance.
>
> By the way, #144 will make sure that every element gets fully initialized before any subsequent macro gets expanded.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/CindyJS/CindyJS/pull/141/files#r47432651


-- 
Michael Strobel, M.Sc.

Technische Universität München -- Zentrum Mathematik
Lehrstuhl für Geometrie und Visualisierung (M10)
Boltzmannstr. 3, 85747 Garching bei München 
https://www-m10.ma.tum.de/bin/view/Lehrstuhl/MichaelStrobel